lib/ok.ex

```diff
@@ -526,4 +526,30 @@ defmodule OK do
       unquote(bind_match(rest) || tmp)
     end
   end
+
+  @doc """
+  Map a function over a list. This function must return a tagged result.
+  If all of the resulting values are tagged :ok, then it returns a list tagged with :ok.
+  If one or more of the resulting values are tagged :error, it returns the first error.
+
+  ## Examples
+
+  iex> OK.mapM([[1, 1], [2, 2], [3, 3]], &safe_div/1)
+  {:ok, [1.0, 1.0, 1.0]}
+
+  iex> OK.mapM([[1, 1], [2, 0], [3, 3]], &safe_div/1)
+  {:error, :zero_division}
+  """
+
+  def mapM(li, fun) do
+    Enum.map(li, fun) |> sequence
```

Comment:
I think this should terminate early if an error is returned.

i.e. a 100 element list which when mapped returns and error on the first item should not process the remaining 99

Reply:
Yes 100%.